server/src/main/java/io/crate/metadata/pgcatalog/PgCatalogTableDefinitions.java

```diff
@@ -197,6 +197,21 @@ public PgCatalogTableDefinitions(InformationSchemaIterables informationSchemaIte
             (user, p) -> p.owner().equals(user.name()),
             PgSubscriptionRelTable.create().expressions()
         ));
+
+        tableDefinitions.put(PgTablesTable.IDENT, new StaticTableDefinition<>(
+            informationSchemaIterables::relations,
+            (user, t) -> user.hasAnyPrivilege(Privilege.Clazz.TABLE, t.ident().fqn())
+                         // we also need to check for views which have privileges set
+                         || user.hasAnyPrivilege(Privilege.Clazz.VIEW, t.ident().fqn()),
+            PgTablesTable.create().expressions()
+        ));
```

Comment:
Afaik `pg_tables` must only contain tables and no views.
```suggestion
        Iterable<TableInfo> tableInfos = () -> InformationSchemaIterables.tablesStream(schemas).iterator();
        tableDefinitions.put(PgTablesTable.IDENT, new StaticTableDefinition<>(
            () -> tableInfos,
            (user, t) -> user.hasAnyPrivilege(Privilege.Clazz.TABLE, t.ident().fqn()),
            PgTablesTable.create().expressions()
        ));
```